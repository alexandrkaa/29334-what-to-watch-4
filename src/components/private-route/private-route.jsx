import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {Route, Redirect} from 'react-router-dom';
import {AppRoutes} from '../../consts/consts.js';
import {hasUserLogined} from '../../reducer/selectors.js';

const PrivateRoute = (props) => {
  const {isAuthorized, render} = props;
  return (
    <Route
      {...props}
      render={(routeProps) => {
        return isAuthorized ? render(routeProps) : <Redirect to={`${AppRoutes.LOGIN_PAGE}`} />;
      }}
    />);
};

PrivateRoute.propTypes = {
  render: PropTypes.func.isRequired,
  isAuthorized: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => ({
  isAuthorized: hasUserLogined(state),
});

export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
