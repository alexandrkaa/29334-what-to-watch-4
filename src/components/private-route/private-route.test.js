import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import {PrivateRoute} from './private-route.jsx';

it(`<PrivateRoute /> should match snapshot`, () => {
  const mockRender = () => {
    return <h1>Header</h1>;
  };

  const tree = renderer
    .create(
        <BrowserRouter>
          <PrivateRoute
            render={mockRender}
            isAuthorized={true}
          />
        </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
