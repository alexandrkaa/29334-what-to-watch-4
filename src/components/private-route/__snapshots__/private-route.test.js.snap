// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> should match snapshot 1`] = `
<h1>
  Header
</h1>
`;
